sympy/core/numbers.py

```diff
@@ -2469,13 +2469,24 @@ def _eval_power(self, expt):
             return S.NaN
         if expt is S.ComplexInfinity:
             return S.NaN
-
-        if expt.is_number:
-            return self**expt.evalf()
+        if expt.is_real is False and expt.is_number:
+            return self._eval_refine(expt)
 
     def _as_mpf_val(self, prec):
         return mlib.finf
 
+    def _eval_refine(self, expt):
```

Comment:
I suspect that there should be only one argument, ``self``, that is a power. Then ``b`` and ``e`` should be derived from it.